modules/apps/dynamic-data-mapping/dynamic-data-mapping-form-field-type/src/main/resources/META-INF/resources/DatePicker/DatePicker.es.js

```diff
@@ -235,32 +286,44 @@ export default function DatePicker({
 		}
 	};
 
+	useEffect(() => {
+		if (datePickerRef?.current) {
+			datePickerRef.current.querySelector(
+				'.date-picker-dropdown-toggle'
+			).onclick = handleDatePickerDropDownClicked;
+		}
+	});
+
 	return (
 		<FieldBase
 			localizedValue={localizedValue}
 			name={name}
 			readOnly={readOnly}
 			{...otherProps}
 		>
-			<ClayDatePicker
-				dateFormat={clayFormat}
-				dir={dir}
-				disabled={readOnly}
-				firstDayOfWeek={firstDayOfWeek}
-				months={months}
-				onBlur={onBlur}
-				onFocus={onFocus}
-				onInput={({target: {value}}) => maskRef.current.update(value)}
-				onValueChange={handleValueChange}
-				placeholder={placeholder}
-				ref={inputRef}
-				time={isDateTime}
-				use12Hours={use12Hours}
-				value={formattedDate}
-				weekdaysShort={weekdaysShort}
-				years={years}
-				yearsCheck={false}
-			/>
+			<div ref={datePickerRef}>
+				<ClayDatePicker
+					dateFormat={clayFormat}
+					dir={dir}
+					disabled={readOnly}
+					firstDayOfWeek={firstDayOfWeek}
+					months={months}
+					onBlur={onBlur}
+					onFocus={onFocus}
+					onInput={({target: {value}}) =>
+						maskRef.current.update(value)
+					}
+					onValueChange={handleValueChange}
+					placeholder={placeholder}
+					ref={inputRef}
+					time={isDateTime}
+					use12Hours={use12Hours}
+					value={formattedDate}
+					weekdaysShort={weekdaysShort}
+					years={years}
+					yearsCheck={false}
+				/>
+			</div>
```

Comment:
I think you supposed do something like that:
```jsx
		const [expanded, setExpanded] = useState(false);
		const handleExpandedChange = (expanded) => {
			if(expanded) {
				onFocus?.();
			}
			else {
				onBlur?.();
			}
		}
```
```suggestion
			<ClayDatePicker
				dateFormat={clayFormat}
				dir={dir}
				disabled={readOnly}
				expanded={expanded}
				firstDayOfWeek={firstDayOfWeek}
				months={months}
				onBlur={onBlur}
				onExpandedChange={handleExpandedChange}
				onFocus={onFocus}
				onInput={({target: {value}}) =>
					maskRef.current.update(value)
				}
				onValueChange={handleValueChange}
				placeholder={placeholder}
				ref={inputRef}
				time={isDateTime}
				use12Hours={use12Hours}
				value={formattedDate}
				weekdaysShort={weekdaysShort}
				years={years}
				yearsCheck={false}
			/>
```